File: Javascript/PalindromicSubstrings.js
// Leetcode Problem - 647

var countSubstrings = function(s){
    let count = 0

    for (let i=0;i<s.length;i++){
        let [l,r] = [i,i]

        while (l>=0 && r<s.length && s[l]===s[r]){
            count+=1
            l-=1
            r+=1
        }

        l = i
        r = i+1

        while (l>=0 && r<s.length && s[l] === s[r]){
            count+=1
            l-=1
            r+=1
        }
    }

    return count
}

s = "abc"
console.log(countSubstrings(s))

File: Python/PalindromicSubstrings.py
## Leetcode Problem - 647

def countSubstrings(s):
    count = 0
    
    for i in range(len(s)):
        l,r = i,i

        while l>=0 and r<len(s) and s[l] == s[r]:
            count +=1
            l-=1
            r+=1

        l,r = i ,i+1
        
        while l>=0 and r<len(s) and s[l] == s[r]:
            count+=1
            l-=1
            r+=1

    return count


s = "abc"
print(countSubstrings(s))